docs/cloud.rst

```diff
@@ -0,0 +1,201 @@
+On the other hand, object storage “scales out” nearly infinitely, meaning that we can make hundreds, thousands, or millions of concurrent reads / writes.
+This makes object storage well suited for distributed data analytics.
+However, the software architecture of a data analysis system must be adapted to take advantage of these properties.
+All large datasets (> 1 GB) in Pangeo Cloud should be stored in Cloud Object Storage.
+
+Reading Data
+^^^^^^^^^^^^
+
+Many pre-existing datasets are browseable at `catalog.pangeo.io <http://catalog.pangeo.io/>`_.
+This catalog is pre-configured to make it easy to open the datasets.
+In most cases, it's as simple as::
+
+   from intake import open_catalog
+   cat = open_catalog("https://raw.githubusercontent.com/pangeo-data/pangeo-datastore/master/intake-catalogs/atmosphere.yaml")
+   ds  = cat["gmet_v1"].to_dask()
+
+To open datasets that are not part of an intake catalog, we recommend using
+the `filesystem-spec <https://filesystem-spec.readthedocs.io/en/latest/>`_ package
+and its related packages `gcsfs <https://gcsfs.readthedocs.io/en/latest/>`_
+(for Google Cloud Storage)
+and `s3fs <https://s3fs.readthedocs.io/en/latest/>`_
+(for Amazon S3 and all S3-compatible object stores).
+
+For example, to open a public file from Google Cloud Storage, you would do::
+
+   import pandas as pd
+   import fsspec
+   path = 'gs://cmip6/cmip6-zarr-consolidated-stores.csv'
+   with fsspec.open(path) as f:
+       df = pd.read_csv(f)
+
+Zarr stores can be opened using ``.get_mapper`` methods from fsspec, gscsfs, and s3fs.
+For examples, see
+
+- `Zarr Docs on Distributed Cloud Storage <https://zarr.readthedocs.io/en/stable/tutorial.html?highlight=s3fs#distributed-cloud-storage>`_
+- `Xarray Docs on Cloud Storage <http://xarray.pydata.org/en/stable/io.html#cloud-storage-buckets>`_
+
+
+Writing Data
+^^^^^^^^^^^^
+
+Writing data (and reading private data) requires credentials for authentication.
+Pangeo Cloud does not provide credentials to individual users.
+Instead you must sign up for your own account with the cloud provider and manage your own storage.
+(Most cloud providers offer several hundred dollars worth of free credits for new accounts.)
+
+On S3-type storage, you will have a client key and client secret associated with you account.
+The following code creates a writeable filesystem::
+
+   fs = s3fs.S3FileSystem(key='<YOUR_CLIENT_KEY>', secret='<YOUR_CLIENT_SECRET')
+
+Non-AWS S3 services (e.g. Wasabi Cloud) can be configured by passing an argument
+such as ``client_kwargs={'endpoint_url': 'https://s3.us-east-2.wasabisys.com'}``
+to ``S3FileSystem``.
+
+For Google Cloud Storage, the best practice is to create a
+`service account <https://cloud.google.com/iam/docs/service-accounts>`_ with
+appropriate permissions to read / write your private bucket.
+You upload your service account key (a .json file) to your Pangeo Cloud
```

Comment:
I hesitate to get in the way of people doing science... but we should think twice about recommending this (even things like storing the S3 keys in cleartext in the notebook).  Jupyterhub admins have access to user's home folders for one.

@yuvipanda do you know if the Berkley team has come up with a way for users to manage and use secrets securely?

Reply:
I hear you. But without this it's impossible to get any work done. Ever since Google blocked the gcsfs "browser" token method (see https://github.com/dask/gcsfs/issues/261), providing service keys is the only way to authenticate. I also need ssh keys in my homedir to access github and other remote services.

I would love to find a more secure solution. But simply not using secrets is not really an option.

Reply:
Just to clarify: +1 to documenting the best approach we have today. I'll poke around on the jupyterhub issue tracker and mailing list to see if this has come up. This sounds like the type of thing that a business might have solved for themselves.

Reply:
- https://discourse.jupyter.org/t/best-practices-for-secrets-in-z2jh-deployments/1292
- https://github.com/berkeley-dsep-infra/datahub/issues/596

Reply:
nm, those are for system secrets, not user secrets.